product-scenarios/scenarios-commons/src/main/java/org/wso2/carbon/esb/scenario/test/common/http/HTTPUtils.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+
+ *      http://www.apache.org/licenses/LICENSE-2.0
+
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.esb.scenario.test.common.http;
+
+import org.apache.axiom.om.OMElement;
+import org.apache.axiom.om.util.AXIOMUtil;
+import org.apache.http.Header;
+import org.apache.http.HeaderElement;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.methods.HttpUriRequest;
+
+import javax.xml.stream.XMLStreamException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Map;
+
+/**
+ * Util class for HTTP Client side functions
+ */
+public class HTTPUtils {
+
+    /**
+     * Returns HTTP_SC code of the response.
+     *
+     * @param response HttpResponse to get status code from
+     * @return  HTTP_SC code as an int
+     */
+    public static int getHTTPResponseCode(HttpResponse response) {
+        return response.getStatusLine().getStatusCode();
+    }
+
+    /**
+     * Extracts the payload from a HTTP response. For a given HttpResponse object, this
+     * method can be called only once.
+     *
+     * @param response HttpResponse instance to be extracted
+     * @return Content payload
+     * @throws IOException If an error occurs while reading from the response
+     */
+    public static String getResponsePayload(HttpResponse response) throws IOException {
+        if (response.getEntity() != null) {
+            InputStream in = response.getEntity().getContent();
+            int length;
+            byte[] tmp = new byte[2048];
+            StringBuilder buffer = new StringBuilder();
+            while ((length = in.read(tmp)) != -1) {
+                buffer.append(new String(tmp, 0, length));
+            }
+            return buffer.toString();
+        }
+        return null;
```

Comment:
Shouldn't we return an emptry string rather than returning null?

Reply:
We need to see when response has no payload response.getEntity() becomes null or not. For now, let's return null to indicate no data is returned?

Reply:
Will return ""